### PR TITLE
Add clients and models for cloud service EPG

### PR DESCRIPTION
**Pull request by gmicol**

### CODE REVIEW

File: models/cloud_svc_epg.go [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+	AzPrivateEndpoint    string `json:",omitempty"`
+	CustomSvcType        string `json:",omitempty"`
+	DeploymentType       string `json:",omitempty"`
+	FloodOnEncap         string `json:",omitempty"`
+	MatchT               string `json:",omitempty"`
+	Name                 string `json:",omitempty"`
+	NameAlias            string `json:",omitempty"`
+	PrefGrMemb           string `json:",omitempty"`
+	Prio                 string `json:",omitempty"`
+	CloudServiceEPg_type string `json:",omitempty"`
```
Comment (sajagana) [inline models/cloud_svc_epg.go:34]: Please check the `exceptionTag` is required.

File: models/cloud_svc_epg.go [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+	A(cloudSvcEPgMap, "azPrivateEndpoint", cloudSvcEPg.AzPrivateEndpoint)
+	A(cloudSvcEPgMap, "customSvcType", cloudSvcEPg.CustomSvcType)
+	A(cloudSvcEPgMap, "deploymentType", cloudSvcEPg.DeploymentType)
+	A(cloudSvcEPgMap, "floodOnEncap", cloudSvcEPg.FloodOnEncap)
+	A(cloudSvcEPgMap, "matchT", cloudSvcEPg.MatchT)
+	A(cloudSvcEPgMap, "name", cloudSvcEPg.Name)
+	A(cloudSvcEPgMap, "nameAlias", cloudSvcEPg.NameAlias)
+	A(cloudSvcEPgMap, "prefGrMemb", cloudSvcEPg.PrefGrMemb)
+	A(cloudSvcEPgMap, "prio", cloudSvcEPg.Prio)
+	A(cloudSvcEPgMap, "CloudServiceEPg_type", cloudSvcEPg.CloudServiceEPg_type)
```
Comment (sajagana) [inline models/cloud_svc_epg.go:68]:
```suggestion
	A(cloudSvcEPgMap, "type", cloudSvcEPg.CloudServiceEPg_type)
```

Is it CloudServiceEPg_type or type?

File: models/cloud_svc_epg.go [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+			AzPrivateEndpoint:    G(CloudServiceEPgCont, "azPrivateEndpoint"),
+			CustomSvcType:        G(CloudServiceEPgCont, "customSvcType"),
+			DeploymentType:       G(CloudServiceEPgCont, "deploymentType"),
+			FloodOnEncap:         G(CloudServiceEPgCont, "floodOnEncap"),
+			MatchT:               G(CloudServiceEPgCont, "matchT"),
+			Name:                 G(CloudServiceEPgCont, "name"),
+			NameAlias:            G(CloudServiceEPgCont, "nameAlias"),
+			PrefGrMemb:           G(CloudServiceEPgCont, "prefGrMemb"),
+			Prio:                 G(CloudServiceEPgCont, "prio"),
+			CloudServiceEPg_type: G(CloudServiceEPgCont, "CloudServiceEPg_type"),
```
Comment (sajagana) [inline models/cloud_svc_epg.go:94]:
```suggestion
			CloudServiceEPg_type: G(CloudServiceEPgCont, "type"),
```
Is it CloudServiceEPg_type or type?

**Review by sajagana — CHANGES_REQUESTED** [review]

File: models/cloud_svc_epg.go (hunk 2 above)
Comment (gmicol) [reply models/cloud_svc_epg.go:68]: you are right. it is "type".

File: models/cloud_svc_epg.go (hunk 3 above)
Comment (gmicol) [reply models/cloud_svc_epg.go:94]: same thing here: it is "type".

File: models/cloud_svc_epselector.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+type CloudServiceEndpointSelector struct {
+	BaseAttributes
+	CloudServiceEndpointSelectorAttributes
+}
+
+type CloudServiceEndpointSelectorAttributes struct {
+	Annotation      string `json:",omitempty"`
+	MatchExpression string `json:",omitempty"`
+	Name            string `json:",omitempty"`
+	NameAlias       string `json:",omitempty"`
```
[the remark this answers is not in the document]
Comment (gmicol) [reply models/cloud_svc_epselector.go:26]: I do not find any of these attributes in the go-client repo, even for other class objects that do require them.

I know we automatically get these attributes in the payload for Ansible ACI but what about the go-client and terraform? @lhercot

**Review by sajagana — APPROVED** [review]
LGTM!

**Review by akinross — APPROVED** [review]
LGTM

**Review by shrsr — APPROVED** [review]
LGTM

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**